test/integration/trackedProgresses.js

```diff
@@ -0,0 +1,457 @@
+  let userId0, userId1, superUser;
+  let userIdToken0, superUserToken;
+  let taskId0, taskId1;
+
+  beforeEach(async function () {
+    userId0 = await addUser(userData0);
+    userIdToken0 = authService.generateAuthToken({ userId: userId0 });
+    userId1 = await addUser(userData1);
+    superUser = await addUser(superUserData);
+    superUserToken = authService.generateAuthToken({ userId: superUser });
+
+    const taskObject0 = await updateTask(taskData[0]);
+    taskId0 = taskObject0.taskId;
+    const taskObject1 = await updateTask(taskData[1]);
+    taskId1 = taskObject1.taskId;
+
+    const docRefUser0 = firestore.collection("trackedProgresses").doc();
+    await docRefUser0.set({ ...predefinedTrackedProgressDataForUser, userId: userId0 });
+
+    const docRefTask0 = firestore.collection("trackedProgresses").doc();
+    await docRefTask0.set({ ...predefinedTrackedProgressDataForTask, taskId: taskId0 });
+  });
+
+  afterEach(async function () {
+    await cleanDb();
+  });
+
+  describe("Verify the POST call for tracked progresses", function () {
+    it("stores the tracked progress document for user", async function () {
+      const response = await chai
+        .request(app)
+        .post("/tracked-progresses")
+        .set("cookie", `${cookieName}=${superUserToken}`)
+        .send({
+          ...trackedProgressUserDataForPost,
+          userId: userId1,
+        });
+      expect(response).to.have.status(201);
+      expect(response.body).to.be.an("object");
+      expect(response.body).to.have.property("message").that.equals("Resource created successfully.");
+      expect(response.body).to.have.property("data").that.is.an("object");
+
+      const data = response.body.data;
+      expect(data).to.have.keys(["id", "type", "userId", "currentlyTracked", "frequency", "createdAt", "updatedAt"]);
+      expect(data).to.have.property("id").that.is.a("string");
+      expect(data).to.have.property("type").that.equals("user");
+      expect(data).to.have.property("userId").that.equals(userId1);
+      expect(data).to.have.property("currentlyTracked").that.equals(true);
+      expect(data).to.have.property("frequency").that.equals(1);
+      expect(data).to.have.property("createdAt").that.is.a("string").and.satisfy(isISOString);
+      expect(data).to.have.property("updatedAt").that.is.a("string").and.satisfy(isISOString);
+    });
+
+    it("throws 409 if tracked progress document already exists for the same user", async function () {
+      const response = await chai
+        .request(app)
+        .post("/tracked-progresses")
+        .set("cookie", `${cookieName}=${superUserToken}`)
+        .send({
+          ...trackedProgressUserDataForPost,
+          userId: userId0,
+        });
+      expect(response).to.have.status(409);
+      expect(response.body).to.be.an("object");
+      expect(response.body).to.have.property("message").that.equals("Resource is already being tracked.");
+    });
```

Comment:
If calling the api again returns 409, how will we remove it from tracking list?

Reply:
No we will not. we need to call patch api with the right payload to change the tracked value.

Reply:
I dont get it which api is that?

Reply:
`/tracked-progresses/:type/:id`

Reply:
This POST api call is to create tracked progress document. if you want to update the frequency and currentlyTracked value you need to make a patch call at this endpoint `/tracked-progresses/:type/:id`